tests/acceptance/features/bootstrap/SecureViewContext.php

```diff
@@ -0,0 +1,332 @@
+				$this->getSession(),
+				AdminAdditionalSettingsPage::HAS_WATERMARK
+			);
+		}
+	}
+
+	/**
+	 * @When the administrator disables print permission in secure view using the webUI
+	 *
+	 * @return void
+	 */
+	public function administratorDisablesCanPrintSecureView() {
+		if ($this->adminAdditionalSettingsPage->isSecureViewOptionEnabled(
+			AdminAdditionalSettingsPage::CAN_PRINT
+		)
+		) {
+			$this->adminAdditionalSettingsPage->toggleSecureViewOption(
+				$this->getSession(),
+				AdminAdditionalSettingsPage::CAN_PRINT
+			);
+		}
+	}
+
+	/**
+	 * @When the administrator disables watermark permission in secure view using the webUI
+	 *
+	 * @return void
+	 */
+	public function administratorDisablesCanWatermarkSecureView() {
+		if ($this->adminAdditionalSettingsPage->isSecureViewOptionEnabled(
+			AdminAdditionalSettingsPage::HAS_WATERMARK
+		)
+		) {
+			$this->adminAdditionalSettingsPage->toggleSecureViewOption(
+				$this->getSession(),
+				AdminAdditionalSettingsPage::HAS_WATERMARK
+			);
+		}
+	}
+
+	/**
+	 * @Then the additional sharing attributes for the response should include
+	 *
+	 * @param TableNode $attributes
+	 *
+	 * @return void
+	 */
+	public function checkingAttributesInLastShareResponse(TableNode $attributes) {
+		$attributes = $attributes->getHash();
+
+		// change string "true"/"false" to boolean inside array
+		\array_walk_recursive(
+			$attributes,
+			function (&$value, $key) {
+				if ($key !== 'enabled') {
+					return;
+				}
+				if ($value === 'true') {
+					$value = true;
+				}
+				if ($value === 'false') {
+					$value = false;
+				}
+			}
+		);
+
+		$responseXml = $this->featureContext->getResponseXml();
+		$actualAttributesElement = $responseXml->xpath('//attributes');
+		if ((bool) $actualAttributesElement) {
+			$actualAttributes = (array) $actualAttributesElement[0];
+			if (empty($actualAttributes)) {
+				throw new Exception(
+					"No data inside 'attributes' element in the last response."
+				);
+			}
+
+			// parse json to array
+			$actualAttributesArray = \json_decode($actualAttributes[0], true);
+			if (\json_last_error() !== JSON_ERROR_NONE) {
+				$errMsg = \strtolower(\json_last_error_msg());
+				throw new Exception(
+					"JSON decoding failed because of $errMsg in json\n" .
+					'Expected data to be json with array of objects. ' .
+					"\nReceived:\n $actualAttributes"
+				);
+			}
+
+			// check if attributes received from table is subset of actualAttributes
+			PHPUnit\Framework\Assert::assertArraySubset(
+				$attributes,
+				$actualAttributesArray
+			);
+			return;
+		}
+
+		throw new Exception("No 'attributes' found inside the response of the last share.");
+	}
+
+	/**
+	 * @Then the downloading of file :fileName for user :user should fail with error message
+	 *
+	 * @param string $fileName
+	 * @param string $user
+	 * @param PyStringNode $errorMessage
+	 *
+	 * @return void
+	 */
+	public function userDownloadsFailWithMessage($fileName, $user, $errorMessage) {
```

Comment:
should that not go into core, or is it a richdocuments specific step?

Reply:
I just did not want to put this in the core, as it would be slow. We could later put this into core, if we feel the need for the step.

Reply:
Please open an issue to move this